arch/c4_radixcore_components.puml

```diff
@@ -0,0 +1,103 @@
+    Component(atoms, "Data Model", "Component: Atom Model", "Radix proprietary UTXO data model.")
+
+    Boundary(utxo_driver, "UTXO Physical Memory Driver") {
+        Component(store_events, "Store Events", "Component: ???")
+        Component(delete_events, "Delete Events", "Component: ???")
+        Component(validation_errors, "Validation Errors", "Component: ???")
+        Component(conflicts, "Conflicts", "Component: Queue???")
+        Component(rt_read_updates, "Real-time Read Updates ???", "Component: ???")
+        Component(rt_data_syncs, "Real-time Data Sync ???", "Component: ???")
+        Component(read_scheduler, "Read Scheduler ???", "Component: ???")
+        Boundary(validate_pipeline, "Validate/Update Pipeline") {
+            Component(utxo_checker, "Stateless UTXO Check", "Component: ???")
+            Component(atomic_state_manager, "Atomic State Manager???", "Component: ???")
+            Rel_(utxo_checker, atomic_state_manager, "uses (valid Atom only)", "-->")
+        }
+        Rel_(store_events, utxo_checker, "uses", "-->")
+        Rel_(delete_events, atomic_state_manager, "uses", "-->")
+        Rel_(utxo_checker, validation_errors, "notifies???", "-->")
+        Rel_(atomic_state_manager, conflicts, "notifies???", "-->")
+        Rel_(atomic_state_manager, rt_read_updates, "notifies???", "-->")
+        Rel_(rt_read_updates, rt_data_syncs, "notifies???", "-->")
+        Rel_(rt_data_syncs, read_scheduler, "read initial state", "-->")
+    }
+
+    Component(spinstate_checker, "Spin State Check", "Component: ???")
+    Component(utxo_virtual_memory, "UTXO Virtualized Memory", "Component: ???")
+
+    Boundary(app_layer, "Application Layer") {
+        Component(app_rules1, "App Rules 1", "Component: Scrypt", "Optimised DSL for Particle specific constraints")
+        Component(app_rules2, "App Rules 2", "Component: Scrypt", "Optimised DSL for Particle specific constraints")
+        Component(cm_manager, "Constraint Machine Manager", "Component: AtomOS", "Compiles *App Rules* into *Contraint Logic Procedures* and manages the life-cycle of Constraint Machines.")
+
+        Boundary(hardware_validation, "Hardware Validation") {
+            Component(cm, "UTXO Hardware", "Component: Constraint Machine", "Hierarchical Finite State Machine (HFSM).")
+            Component(clp1, "Constraint Logic Procedure 1", "Component: Dynamic", "Compiled HFSM guard condition (from App Rule 1).")
+            Component(clp2, "Constraint Logic Procedure 2", "Component: Dynamic", "Compiled HFSM guard condition (from App Rule 2).")
+
+            Rel_(cm, clp1, "evaluates valid UTXO", "-->")
+            Rel_(cm, clp2, "evaluates valid UTXO", "-->")
+        }
+
+        Rel_(app_rules1, cm_manager, "registers with", "-->")
+        Rel_(app_rules2, cm_manager, "registers with", "-->")
+        Rel_(cm_manager, clp1, "compiles???", "-->")
+        Rel_(cm_manager, clp2, "compiles???", "-->")
```

Comment:
"compiles" is a strong word here. I think "translates" makes more sense for now.